### PR TITLE
Use zod for config validation

### LINKED ISSUE #88: Use zod for config validation

### PR DESCRIPTION
**Pull request by imMohika**
Closes #88

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,26 +1,48 @@
 import "dotenv/config";
-import { cleanEnv, json, num, str } from "envalid";
```
Comment (deptyped) [inline src/config.ts:2]: Please remove envalid also from the dependencies list

File: src/config.ts [hunk 2]
```diff
@@ -1,26 +1,48 @@
...
+  ),
+  BOT_TOKEN: z.string(),
+  BOT_WEBHOOK: z.string().url(),
+  BOT_ADMIN_USER_ID: z.coerce.number().finite(),
 });
 
-export type Config = typeof config;
+export const config = configSchema.parse(process.env);
+export const isDev = config.NODE_ENV === "development";
+export const isProd = config.NODE_ENV === "production";
```
Comment (deptyped) [inline src/config.ts:47]: Can we have these variables inside the config schema?

```suggestion
  isDev: z.boolean(),
  isProd: z.boolean(),
});

export const config = configSchema.parse({
  ...process.env,
  isDev: process.env.NODE_ENV === "development",
  isProd: process.env.NODE_ENV === "production",
})
```

**Review by deptyped — CHANGES_REQUESTED** [review]

File: src/config.ts (hunk 2 above)
Comment (imMohika) [reply src/config.ts:47]: How about adding them inside a parseConfig function?
```typescript
const parseConfig = (env: NodeJS.ProcessEnv) => {
  const config = configSchema.parse(env);

  return {
    ...config,
    isDev: config.NODE_ENV === "development",
    isProd: config.NODE_ENV === "production",
  };
};

export const config = parseConfig(process.env);
```

Comment (deptyped) [reply src/config.ts:47]: Looks good to me

File: src/logger.ts [hunk 3]
```diff
@@ -5,14 +5,15 @@ export const createLogger = (config: Config) => {
...
+  const isDev = config.NODE_ENV === "development";
 
   const transport = pino.transport({
     targets: [
       {
         target: "pino-pretty",
         level: config.LOG_LEVEL,
         options: {
-          ...(config.isDev && {
+          ...(isDev && {
```
Comment (deptyped) [inline src/logger.ts:16]:
```suggestion
  const transport = pino.transport({
    targets: [
      {
        target: "pino-pretty",
        level: config.LOG_LEVEL,
        options: {
          ...(config.isDev && {
```

**Review by deptyped — CHANGES_REQUESTED** [review]

**Comment by deptyped** [comment]
Can you please run `npm i` command and commit `package-lock.json` changes to complete a CI check?

**Review by deptyped — APPROVED** [review]
Looks good to me. Thanks for the contribution!

**Pull request merged**